File: ios/HelloWorld.m
//
//  HelloWorld.m
//  bridgeee
//
//  Created by Muhammad Mukthayar on 25/05/2024.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(HelloWorld,NSObject)
RCT_EXTERN_METHOD(helloworld)
RCT_EXTERN_METHOD(sumNumbers : (NSInteger)a withB:(NSInteger)b resolver:(RCTPromiseResolveBlock)resolver rejecter:(RCTPromiseRejectBlock)rejecter)

@end

File: ios/HelloWorld.swift
//
//  HelloWorld.swift
//  bridgeee
//
//  Created by Muhammad Mukthayar on 25/05/2024.
//

import Foundation
import React


@objc(HelloWorld)
class HelloWorld : NSObject {
  
  @objc
  func helloworld(){
    print("Mukthayar is here...")
  }
  
  @objc(sumNumbers:withB:resolver:rejecter:)
  func sumNumbers(a: Int, b: Int, resolver: RCTPromiseResolveBlock, rejecter: RCTPromiseRejectBlock) {
      let result = a + b
        if(result > 0){
            resolver(result)
          }else{
            rejecter("NEGATIVITY NOT ALLOW HEREEE","sss",nil)
          }
      
  }
}
